### PR TITLE
Update duct related calcs in airflow.rb

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Addresses https://trello.com/c/VmcGehGM/62-duct-leakage-total.

Based on ASHRAE Standard 152:

Change equations for determining fraction of ducts outside conditioned space:
```
  def self.get_location_frac_leakage(location_frac, stories)
    if location_frac == Constants.Auto
      if stories == 1
        location_frac_leakage = 1
      else
        location_frac_leakage = 0.75
      end
    else
      location_frac_leakage = location_frac.to_f
    end
    return location_frac_leakage
  end
```

Change equations for determining supply/return surface areas:
```
  def self.get_duct_supply_surface_area(ffa)
    return 0.27 * ffa # ft^2
  end
```
```
  def self.get_duct_return_surface_area(ffa, num_returns)
    if num_returns < 6
      b_r = 0.05 * num_returns
    else
      b_r = 0.25
    end
    return b_r * ffa # ft^2
  end
```

Multiply duct areas by the fraction of ducts outside conditioned space:
```
    supply_surface_area = Airflow.get_duct_supply_surface_area(unit_ffa) * ducts.supply_area_mult * f_out
    return_surface_area = Airflow.get_duct_return_surface_area(unit_ffa, ducts.num_returns) * ducts.return_area_mult * f_out
```

Then determine supply/return ua by:
```
      supply_ua = supply_surface_area / supply_r
      return_ua = return_surface_area / return_r
```
which is used in the ducts ems subroutine.

Finally, determine supply/return loss by:
```
      supply_loss = f_out * ( supply_leakage + ah_supply_leakage )
      return_loss = f_out * ( return_leakage + ah_return_leakage )
```
which is used by sizing.rb.

## Checklist

Not all may apply:

- [ ] Unit tests have been added or updated
- [ ] All rake tasks have been run, and pass
- [ ] Documentation has been modified appropriately
- [ ] Any new options are added to `project_testing`
- [ ] `project_testing` runs without any failures
- [ ] No unexpected regression test changes
- [ ] All tests are passing
- [ ] The [changelog](https://github.com/NREL/resstock/blob/master/CHANGELOG.md) has been updated appropriately
- [ ] This branch is up-to-date with develop

For more information on how to perform these checklist items, see the documentation's [Advanced Tutorial](https://resstock.readthedocs.io/en/latest/advanced_tutorial/index.html).

### CODE REVIEW

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb [hunk 1]
```diff
@@ -1113,28 +1113,26 @@ def self.process_ducts_for_unit(model, runner, ducts, building, unit, unit_index
 
     location_frac_conduction = location_frac_leakage
     ducts.num_returns = Airflow.get_num_returns(ducts.num_returns, num_stories)
-    supply_surface_area = Airflow.get_duct_supply_surface_area(ducts.supply_area_mult, unit_ffa, num_stories)
-    return_surface_area = Airflow.get_return_surface_area(ducts.return_area_mult, unit_ffa, num_stories, ducts.num_returns)
+    supply_surface_area = Airflow.get_duct_supply_surface_area(unit_ffa) * ducts.supply_area_mult
+    return_surface_area = Airflow.get_duct_return_surface_area(unit_ffa, ducts.num_returns) * ducts.return_area_mult
```
Comment (ejhw) [inline resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:1117]: Is this because Std 152 has a different method of calculating surface areas than BAHSP?

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb [hunk 2]
```diff
@@ -1113,28 +1113,26 @@ def self.process_ducts_for_unit(model, runner, ducts, building, unit, unit_index
...
 
     # Calculate Duct UA value
     if location_name != unit_living.zone.name.to_s
-      unconditioned_duct_area = supply_surface_area * location_frac_conduction
       supply_r = Airflow.get_duct_insulation_rvalue(ducts.r, true)
       return_r = Airflow.get_duct_insulation_rvalue(ducts.r, false)
-      unconditioned_ua = unconditioned_duct_area / supply_r
+      supply_ua = supply_surface_area / supply_r
       return_ua = return_surface_area / return_r
```
Comment (ejhw) [inline resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:1124]: It looks like this change means that neither supply nor return ua would use the location_frac. Instead I think we want both of them to use location_frac.

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb [hunk 3]
```diff
@@ -2243,11 +2241,10 @@ def self.calc_duct_leakage_from_test(ducts, ffa, fan_AirFlowRate)
 
   def self.get_location_frac_leakage(location_frac, stories)
     if location_frac == Constants.Auto
-      # Duct location fraction per 2010 BA Benchmark
       if stories == 1
         location_frac_leakage = 1
       else
-        location_frac_leakage = 0.65
+        location_frac_leakage = 0.75
```
Comment (ejhw) [inline resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:2247]: Add comment citing new source of fraction (hpxml docs? and/or Std 152)

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb [hunk 4]
```diff
@@ -2308,22 +2305,17 @@ def self.get_duct_insulation_rvalue(nominalR, isSupply)
     end
   end
 
-  def self.get_duct_supply_surface_area(mult, ffa, num_stories)
-    # Duct Surface Areas per 2010 BA Benchmark
```
Comment (ejhw) [inline resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:2312]: Add comment citing new source

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb [hunk 5]
```diff
@@ -2308,22 +2305,17 @@ def self.get_duct_insulation_rvalue(nominalR, isSupply)
...
-      return 0.27 * ffa * mult # ft^2
-    else
-      return 0.2 * ffa * mult
-    end
+  def self.get_duct_supply_surface_area(ffa)
+    return 0.27 * ffa # ft^2
   end
 
-  def self.get_return_surface_area(mult, ffa, num_stories, num_returns)
-    # Duct Surface Areas per 2010 BA Benchmark
```
Comment (ejhw) [inline resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:2321]: Add comment citing new source

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb (hunk 1 above)
Comment (joseph-robertson) [reply resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:1117]: I'm getting it from the OS-HPXML repo: https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/hvac.rb#L4017-L4031

**Comment by shorowit** [comment]
I ran a couple buildings and compared duct areas to OS-HPXML. Supply areas match up perfectly, but return duct areas are off because of different default assumptions about number of return registers. Resstock is using [1+NumStories](https://github.com/NREL/resstock/blob/a1737ab6caf2367d3c3bf15a1be3baa579582438/resources/measures/HPXMLtoOpenStudio/resources/airflow.rb#L2334) (based on Building America) while OS-HPXML is using [the number of conditioned floors](https://openstudio-hpxml.readthedocs.io/en/latest/workflow_inputs.html#air-distribution) (based on ASHRAE 152). So for a 1-story home on slab, ResStock has 2x the return duct area as OS-HPXML.

I haven't yet looked at duct leakage

Also, note that the duct area cost multiplier is called "Duct Surface Area (ft^2)", though it's actually the duct surface area outside conditioned space. Not sure if that should be clarified/documented somewhere.

**Comment by shorowit** [comment]
Hmm. Looking over the code, I suspect there may also be a difference in duct areas for homes w/ conditioned basements. I will check that.

File: resources/measures/HPXMLtoOpenStudio/resources/airflow.rb [hunk 6]
```diff
@@ -1108,33 +1108,30 @@ def self.process_ducts_for_unit(model, runner, ducts, building, unit, unit_index
       ah_return_leakage = ducts.ah_return_frac * ducts.total_leakage
     end
 
-    # Fraction of ducts in primary duct location (remaining ducts are in above-grade conditioned space).
-    location_frac_leakage = Airflow.get_location_frac_leakage(ducts.location_frac, num_stories)
-
-    location_frac_conduction = location_frac_leakage
+    # Calculate duct areas
+    # Based on ASHRAE Standard 152 (https://www.energy.gov/eere/buildings/downloads/ashrae-standard-152-spreadsheet)
+    f_out = Airflow.get_location_frac_leakage(ducts.location_frac, num_stories) # Fraction of ducts in primary duct location (remaining ducts are in above-grade conditioned space).
```
Comment (shorowit) [inline resources/measures/HPXMLtoOpenStudio/resources/airflow.rb:1113]: For consistency with ASHRAE 152, you should be passing in `building.stories`, not `num_stories`, as the logic is based on the [number of conditioned floors above-grade](https://github.com/NREL/OpenStudio-HPXML/blob/ffbc9fdae4cb16fac8b9bddabb787ef883ecceee/HPXMLtoOpenStudio/resources/hvac.rb#L4019).

**Review by shorowit — APPROVED** [review]
The duct calculations look correct to me now.

**Comment by ejhw** [comment]
> Also, note that the duct area cost multiplier is called "Duct Surface Area (ft^2)", though it's actually the duct surface area outside conditioned space. Not sure if that should be clarified/documented somewhere. EDIT: That's not quite right. It's actually the duct surface area outside _above-grade conditioned space_, which could be conditioned in the case of a conditioned basement or unconditioned in the case of an unconditioned basement, crawlspace, or unconditioned attic.

Thanks @shorowit. Would it be possible to change the cost multiplier to exclude duct area in conditioned basements?
